### PR TITLE
refactor circle to ellipse

### LINKED ISSUE #2694: Change the name of "Draw Circle" to "Draw Ellipse"
**Is your feature request related to a problem? Please describe.**
It's more accurate to what it does. Follows the same idea as how "Draw Rectangle" is used to draw general rectangles as well as squares which are just special cases of rectangles. A circle is just a special case of an ellipse, so all ellipses shouldn't be called circles. 

**Describe the solution you'd like**
Change the name of the "Draw Circle" tool to "Draw Ellipse".

### PR DESCRIPTION
**Pull request by Baseng0815**
Closes #2694.

I changed labels as well as internals like tool and class names. Seems to work fine. Translations for other languages still need to be changed.

### CODE REVIEW

**Comment by rolandlo** [comment]
You don't need to(and shouldn't) change .po-files. This will be handled in the Crowdin project.
Make sure that Clang format is applied, though.

File: src/control/tools/EllipseHandler.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ void CircleHandler::drawShape(Point& c, const PositionInputData& pos) {
             this->modifyModifiersByDrawDir(width, height, true);
         }
 
-        if (this->modShift)  // make circle
+        if (this->modShift)  // make ellipse
```
Comment (rolandlo) [inline src/control/tools/EllipseHandler.cpp:44]: here it does make a circle, so don't change the comment to "make ellipse"

File: src/plugin/luapi_application.h [hunk 2]
```diff
@@ -164,7 +164,7 @@ static int applib_registerUi(lua_State* L) {
  * Example 1: app.uiAction({["action"] = "ACTION_PASTE"})
  * pastes the clipboard content into the document
  *
- * Example 2: app.uiAction({["action"] = "ACTION_TOOL_DRAW_CIRCLE", ["enabled"] = false})
+ * Example 2: app.uiAction({["action"] = "ACTION_TOOL_DRAW_ELLIPSE", ["enabled"] = false})
  * turns off the Circle drawing type
```
Comment (rolandlo) [inline src/plugin/luapi_application.h:168]:
```suggestion
 * turns off the Ellipse drawing type
```

File: src/control/ToolEnums.cpp [hunk 3]
```diff
@@ -172,8 +172,8 @@ auto toolTypeFromString(const string& type) -> ToolType {
     if (type == "drawRect") {
         return TOOL_DRAW_RECT;
     }
-    if (type == "drawCircle") {
-        return TOOL_DRAW_CIRCLE;
+    if (type == "drawEllipse") {
```
Comment (rolandlo) [inline src/control/ToolEnums.cpp:175]:
```suggestion
    // recognize previous spelling of Highlighter, V1.1.0+dev (Jan 2021) and earlier
    if (type == "drawEllipse" || type == "drawCircle") {
```

**Review by rolandlo — CHANGES_REQUESTED** [review]
I took the time to look at your PR more closely. Some more changes are needed:
- change `DRAW_CIRCLE` to `DRAW_ELLIPSE` in `toolbar.ini`. Otherwise the toolbars can't load the circle tools.
- In `ToolMenuHandler::load` add the lines
```c++                
                // recognize previous name, V1.1.0+dev (Jan 2021) and earlier
                if (name == "DRAW_CIRCLE") {
                    name = "DRAW_ELLIPSE";
               }
```
so that custom toolbars that are not changed through this PR continue to work as well

- Replace `CircleHandler` by `EllipseHandler` in the comment on line 78 in `SplineHandler.h`

File: src/control/ToolEnums.cpp (hunk 3 above)
Comment (Baseng0815) [reply src/control/ToolEnums.cpp:175]: Don't you mean
`// recognize previous spelling of Ellipse, V1.1.0+dev (Jan 2021) and earlier`
?

Comment (rolandlo) [reply src/control/ToolEnums.cpp:175]: Yes, sorry for the typo. You can amend it to the current one.

**Comment by rolandlo** [comment]
You will also need to rebase on current master and potentially fix a merge conflict in `CircleHandler.cpp`

**Review by rolandlo — APPROVED** [review]
LGTM. Can I merge or did you still want to change something?

**Comment by Baseng0815** [comment]
> LGTM. Can I merge or did you still want to change something?

You can merge.

**Comment by rolandlo** [comment]
Ok, thanks for your contribution!

**Pull request merged**